commit 7a9fde9bc14334c9ee2598f5dc9ea3477d317ad2
Author: Richard Hughes <richard@hughsie.com>
Date:   Thu Apr 29 09:10:52 2010 +0100

    Fix an update viewer crash if the GetUpdates call produced any messages

diff --git a/src/gpk-update-viewer.c b/src/gpk-update-viewer.c
index 29c308d..98d1a9c 100644
--- a/src/gpk-update-viewer.c
+++ b/src/gpk-update-viewer.c
@@ -2598,7 +2598,7 @@ gpk_update_viewer_get_updates_cb (PkClient *client, GAsyncResult *res, GpkUpdate
 	/* do we have any important messages we need to show? */
 	array_messages = pk_results_get_message_array (results);
 	for (i=0; i<array_messages->len; i++) {
-		message = g_ptr_array_index (array, i);
+		message = g_ptr_array_index (array_messages, i);
 		g_object_get (message,
 			      "type", &message_type,
 			      NULL);
